src/wp-includes/block-editor.php

```diff
@@ -286,28 +286,33 @@ function get_legacy_widget_block_editor_settings() {
  * Collect the block editor assets that need to be loaded into the editor's iframe.
  *
  * @since 6.0.0
+ * @since 7.0.0 Allow enqueuing script modules.
  * @access private
  *
- * @global WP_Styles  $wp_styles  The WP_Styles current instance.
- * @global WP_Scripts $wp_scripts The WP_Scripts current instance.
+ * @global WP_Styles        $wp_styles          The WP_Styles current instance.
+ * @global WP_Scripts       $wp_scripts         The WP_Scripts current instance.
+ * @global WP_Script_Modules $wp_script_modules The WP_Script_Modules current instance.
  *
  * @return array {
  *     The block editor assets.
  *
- *     @type string|false $styles  String containing the HTML for styles.
- *     @type string|false $scripts String containing the HTML for scripts.
+ *     @type string|false $styles         String containing the HTML for styles.
+ *     @type string|false $scripts        String containing the HTML for scripts.
+ *     @type string|false $script_modules String containing the HTML for scripts.
  * }
  */
 function _wp_get_iframed_editor_assets() {
-	global $wp_styles, $wp_scripts;
+	global $wp_styles, $wp_scripts, $wp_script_modules;
 
 	// Keep track of the styles and scripts instance to restore later.
-	$current_wp_styles  = $wp_styles;
-	$current_wp_scripts = $wp_scripts;
+	$current_wp_styles         = $wp_styles;
+	$current_wp_scripts        = $wp_scripts;
+	$current_wp_script_modules = $wp_script_modules;
 
 	// Create new instances to collect the assets.
 	$wp_styles  = new WP_Styles();
 	$wp_scripts = new WP_Scripts();
+	$wp_script_modules = $wp_script_modules->clone_without_enqueued_modules();
```

Comment:
I learned of a hack/feature in PHP to allow setting private members [in PostStatus Slack](https://poststatus.slack.com/archives/C0413TGQZ/p1763032090544669). This should should have the same effect, including my [suggestion](https://github.com/WordPress/wordpress-develop/pull/10597/files#r2603689193) below:

```suggestion
	$wp_script_modules = clone $wp_script_modules;
	( function () {
		$this->done  = array();
		$this->queue = array();
	} )->call( $wp_script_modules );
```

Reply:
That's wild. I can't believe it works and private properties are accessible in the closure.

Reply:
That's a clever trick, but I'm not comfortable actually using it in Core and would prefer to keep the clone behavior in a method.